lib/archethic/contracts/interpreter/transaction_statements.ex

```diff
@@ -175,9 +179,9 @@ defmodule Archethic.Contracts.Interpreter.TransactionStatements do
 
     ownership =
       Ownership.new(
-        Utils.maybe_decode_hex(secret),
-        Utils.maybe_decode_hex(secret_key),
-        Enum.map(authorized_public_keys, &Utils.maybe_decode_hex(&1))
+        SCUtils.maybe_decode_hex(secret),
+        SCUtils.maybe_decode_hex(secret_key),
+        Enum.map(authorized_public_keys, &SCUtils.maybe_decode_hex(&1))
```

Comment:
We could also verify if public keys are correct using `Crypto.valid_public_key?`